From c263cabec40901ce653bc1de8e1235adda1b855b Mon Sep 17 00:00:00 2001
From: Mario Ivankovits <imario@apache.org>
Date: Mon, 14 May 2007 18:19:47 +0000
Subject: [PATCH] VFS-144: Use the first registered LocalFileProvider. This
 follows the VFS rules of not allowing to override already registered
 providers - Thanks to Adam Heath for the patch!

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/vfs/trunk@537931 13f79535-47bb-0310-9956-ffa450edef68
---
 .../vfs/impl/DefaultFileSystemManager.java    | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/core/src/main/java/org/apache/commons/vfs/impl/DefaultFileSystemManager.java b/core/src/main/java/org/apache/commons/vfs/impl/DefaultFileSystemManager.java
index 63f9b4e243..7214532c94 100644
--- a/core/src/main/java/org/apache/commons/vfs/impl/DefaultFileSystemManager.java
+++ b/core/src/main/java/org/apache/commons/vfs/impl/DefaultFileSystemManager.java
@@ -16,17 +16,6 @@
  */
 package org.apache.commons.vfs.impl;
 
-import java.io.File;
-import java.net.URLStreamHandler;
-import java.net.URLStreamHandlerFactory;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.lang.reflect.Constructor;
-
 import org.apache.commons.logging.Log;
 import org.apache.commons.vfs.CacheStrategy;
 import org.apache.commons.vfs.FileContentInfoFactory;
@@ -53,6 +42,17 @@
 import org.apache.commons.vfs.provider.UriParser;
 import org.apache.commons.vfs.provider.VfsComponent;
 
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.net.URLStreamHandler;
+import java.net.URLStreamHandlerFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
 /**
  * A default file system manager implementation.
  *
@@ -186,7 +186,7 @@ public void addProvider(final String[] urlSchemes,
 			providers.put(scheme, provider);
 		}
 
-		if (provider instanceof LocalFileProvider)
+		if (provider instanceof LocalFileProvider && localFileProvider == null)
 		{
 			localFileProvider = (LocalFileProvider) provider;
 		}
